src/fs/unix.ts

```diff
@@ -0,0 +1,203 @@
+ */
+export async function insertNodeIntoTree(
+  node: PBNode,
+  path: Path.Distinctive<Path.Segments>,
+  depot: Depot.Implementation,
+  fileCID?: CID
+): Promise<PBNode> {
+  const pathKind = Path.kind(path)
+  const pathParts = Path.unwrap(path)
+  const name = pathParts[0]
+  const link = node.Links.find(l => l.Name === name)
+
+  // Directory
+  // ---------
+  if (Path.length(path) > 1) {
+    let dirNode: PBNode
+
+    if (link?.Hash) {
+      dirNode = await load(link.Hash, depot)
+    } else {
+      dirNode = createDirectory(new Date())
+    }
+
+    const newDirNode = await insertNodeIntoTree(
+      dirNode,
+      Path.fromKind(pathKind, ...pathParts.slice(1)),
+      depot,
+      fileCID
+    )
+
+    const dirCID = await store(newDirNode, depot)
+    const links = link
+      ? replaceLinkHash(node.Links, name, dirCID)
+      : addLink(node.Links, name, dirCID)
+
+    return { ...node, Links: links }
+  }
+
+  // Last part of path
+  // -----------------
+  // Directory
+  if (pathKind === "directory") {
+    if (link) return node
+
+    const dirNode = createDirectory(new Date())
+    const dirCID = await store(dirNode, depot)
+
+    const links = addLink(node.Links, name, dirCID)
+    return { ...node, Links: links }
+  }
+
+  // File
+  if (!fileCID) throw new Error("Need a file CID when adding a UnixFS file")
+
+  const links = link
+    ? replaceLinkHash(node.Links, name, fileCID)
+    : addLink(node.Links, name, fileCID)
+
+  return { ...node, Links: links }
+}
+
+/**
+ * Load a UnixFS node.
+ */
+export async function load(cid: CID, depot: Depot.Implementation): Promise<PBNode> {
+  return DagPB.decode(
+    await depot.getBlock(cid)
+  )
+}
+
+/**
+ * Remove a node from a UnixFS tree.
+ */
+export async function removeNodeFromTree(
+  node: PBNode,
+  path: Path.Distinctive<Path.Segments>,
+  depot: Depot.Implementation
+): Promise<PBNode> {
+  const pathKind = Path.kind(path)
+  const pathParts = Path.unwrap(path)
+  const name = pathParts[0]
+  const link = node.Links.find(l => l.Name === name)
+
+  // Directory
+  // ---------
+  if (Path.length(path) > 1) {
+    let dirNode: PBNode
+
+    if (link?.Hash) {
+      dirNode = await load(link.Hash, depot)
+    } else {
+      return node
+    }
+
+    const newDirNode = await removeNodeFromTree(
+      dirNode,
+      Path.fromKind(pathKind, ...pathParts.slice(1)),
+      depot
+    )
+
+    const dirCID = await store(newDirNode, depot)
+    const links = link
+      ? replaceLinkHash(node.Links, name, dirCID)
+      : addLink(node.Links, name, dirCID)
+
+    return { ...node, Links: links }
+  }
+
+  // Last part of path
+  // -----------------
+  if (!link) return node
+  return { ...node, Links: node.Links.filter(l => l.Name !== name) }
+}
+
+/**
+ * Store a UnixFS node.
+ */
+export async function store(node: PBNode, depot: Depot.Implementation): Promise<CID> {
+  return depot.putBlock(
+    DagPB.encode(node),
+    DagPB.code
+  )
+}
+
+////////
+// ㊙️ //
```

Comment:
I still find the emoji section headers a bit confusing 🙈 after some searching, it looks like this means secret? ㊙️ Should we just add the word `Secret` to these sections? May be clearer for outside contributors 🤔

Reply:
Ah right I promised you that emoji guide, I'll do that in this PR. I like that the un-exported items are grouped together in a section that is color coded by a red emoji 😅 (and conveniently it's the sign for "secret" yeah). I can also append the word secret after the emoji? 🤷

Reply:
haha no worries, yeah I think just including the word `Secret` next to the emoji would be helpful, rather than having an emoji guide to refer to. I'm open to anything though. Whatever the most clear solution is for outside developers to tell what the section refers to 👍🏼

Reply:
Oh wait I did already add this, it's in `src/README.md`. It doesn't have an index table though, I'll add that.